Brokerages/Oanda/OandaBrokerageFactory.cs

```diff
@@ -0,0 +1,174 @@
+            /// </summary>
+            public static string AccessToken
+            {
+                get { return Config.Get("oanda-access-token"); }
+            }
+
+
+            /// <summary>
+            /// Gets the access token from configuration
+            /// </summary>
+            public static string Environment
+            {
+                get { return Config.Get("oanda-environment"); }
+            }
+        }
+        /// <summary>
+        /// Initializes a new instance of the <see cref="BrokerageFactory"/> class.
+        /// </summary>
+        public OandaBrokerageFactory() : base(typeof(OandaBrokerage))
+        {
+        }
+
+        /// <summary>
+        /// Performs application-defined tasks associated with freeing, releasing, or resetting unmanaged resources.
+        /// </summary>
+        public override void Dispose()
+        {
+        }
+
+
+        /// <summary>
+        /// File path used to store tradier token data
+        /// </summary>
+        public const string TokensFile = "oanda-tokens.txt";
+
+        /// <summary>
+        /// Gets the brokerage data required to run the brokerage from configuration/disk
+        /// </summary>
+        /// <remarks>
+        /// The implementation of this property will create the brokerage data dictionary required for
+        /// running live jobs. See <see cref="IJobQueueHandler.NextJob"/>
+        /// </remarks>
+        public override Dictionary<string, string> BrokerageData
+        {
+            get
+            {
+                string accessToken;
+
+                var accountId = Configuration.AccountId.ToString();
+                var data = new Dictionary<string, string>();
+                if (File.Exists(TokensFile))
+                {
+                    var tokens = JsonConvert.DeserializeObject<TokenResponse>(File.ReadAllText(TokensFile));
+                    accessToken = tokens.AccessToken;
+                } 
+                else
+                {
+                    accessToken = Configuration.AccessToken;
+                }
+                data.Add("oanda-account-id", accountId);
+                data.Add("oanda-access-token", accessToken);
+                return data;
+            }
+        }
+
+        /// <summary>
+        /// Creates a new IBrokerage instance
+        /// </summary>
+        /// <param name="job">The job packet to create the brokerage for</param>
+        /// <param name="algorithm">The algorithm instance</param>
+        /// <returns>A new brokerage instance</returns>
+        public override IBrokerage CreateBrokerage(LiveNodePacket job, IAlgorithm algorithm)
+        {
+            var errors = new List<string>();
+            var accountId = Read<int>(job.BrokerageData, "oanda-account-id", errors);
+            var accessToken = Read<string>(job.BrokerageData, "oanda-access-token", errors);
+            var refreshToken = Read<string>(job.BrokerageData, "oanda-refresh-token", errors);
```

Comment:
the refresh token is only being used for communication with /processOanda, I think this can be safely removed